main.go

```diff
@@ -384,13 +417,17 @@ func newController(klient kubernetes.Interface, logger log.Logger, o *options) *
 func (c *controller) registerMetrics(reg *prometheus.Registry) {
 	if reg != nil {
 		c.reconcileAttempts.Add(0)
-		c.reconcileErrors.WithLabelValues(fetch).Add(0)
-		c.reconcileErrors.WithLabelValues(decode).Add(0)
-		c.reconcileErrors.WithLabelValues(save).Add(0)
+		c.reconcileErrors.WithLabelValues(fetchLabel).Add(0)
+		c.reconcileErrors.WithLabelValues(decodeLabel).Add(0)
+		c.reconcileErrors.WithLabelValues(saveLabel).Add(0)
 		c.configmapChangeAttempts.Add(0)
-		c.configmapChangeErrors.WithLabelValues(create).Add(0)
-		c.configmapChangeErrors.WithLabelValues(update).Add(0)
-		c.configmapChangeErrors.WithLabelValues(other).Add(0)
+		c.configmapChangeErrors.WithLabelValues(createLabel).Add(0)
+		c.configmapChangeErrors.WithLabelValues(updateLabel).Add(0)
+		c.configmapChangeErrors.WithLabelValues(otherLabel).Add(0)
+		c.cleanupErrors.WithLabelValues(createLabel).Add(0)
+		c.cleanupErrors.WithLabelValues(decodeLabel).Add(0)
+		c.cleanupErrors.WithLabelValues(deleteLabel).Add(0)
+		c.cleanupErrors.WithLabelValues(fetchLabel).Add(0)
```

Comment:
Maybe it's worth looping bucks over these labels?

Reply:
the thing is that not every counter uses every label, it's a bit funny. we'd end up with three different loops :/